Add some entry level typing for the triggers.create endpoint

shapirone opened the pull request:
###  Summary
This introduces introductory typing for the `triggers.create` method and adds some architecture to support custom typing for individual methods.

Future PRs will add:
- More specific typing for each method
- Filters
- Responses

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/deno-slack-api/blob/main/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

shapirone commented on src/typed-method-types/workflows/triggers/mod.ts:
```diff
@@ -0,0 +1,44 @@
+import { ShortcutTrigger } from "./shortcut.ts";
+import { WebhookTrigger } from "./webhook.ts";
+
+// TODO: Export to SDK
+export const VALID_TRIGGERS = {
+  event: "event",
+  scheduled: "scheduled",
+  shortcut: "shortcut",
+  webhook: "webhook",
+} as const;
```
This is an area where I'd love other peoples opinions on how we should surface these valid triggers in the SDK. As written, the only way these will be surfaced to the SDK is via typeahead. This is going to come up again for a few different parts of triggers.

I think that most users wouldn't expect to import these from the API Client and instead expect to import from the SDK, but importing these to the SDK just to export them again introduces a dependency on the API client.

Options:
- Introduce the dependency and import the API library in the SDK
- Solve discoverability through documentation and keep exports in the API library
- Do nothing and lean on typeahead

shapirone replied:
> I assume we'll need to have the SDK depend on the API library to implement that hook.

FWIW since the developer is working out of their own project which already includes both libraries both options are viable, it's just be a matter of what library they import into their trigger definition file:
```ts
import { TriggerTypes } from "deno-slack-sdk/mod.ts";
// vs
import { TriggerTypes } from "deno-slack-api/mod.ts";
```

mcodik reviewed (approved):
I left a few comments, tho the structures look generally right to me. not super familiar with typescript types, so may want to depend on others for feedback on that :)

shapirone commented on src/typed-method-types/workflows/triggers/mod.ts:
```diff
@@ -0,0 +1,44 @@
+  description?: string;
+};
+
+// A helper to make sure inputs are passed. Required for automated triggers
```
[It looks like `inputs` is still an optional param for `triggers.create`](https://slack-github.com/slack/webapp/blob/master/api_metadata/methods/workflows.triggers/create.api#L14)

shapirone commented on src/typed-method-types/workflows/triggers/shortcut.ts:
```diff
@@ -0,0 +1,9 @@
+export type ShortcutTrigger = BaseTrigger & {
+  type: typeof TriggerTypes.Shortcut;
+  shortcut?: {
+    // Eventually this will support button text
```
Nice, I'll add that!

filmaj commented on src/typed-method-types/typed-method-tests.ts:
```diff
@@ -0,0 +1,12 @@
+import { assertEquals } from "https://deno.land/std@0.99.0/testing/asserts.ts";
+import { SlackAPI } from "../mod.ts";
+
+Deno.test("Custom Type Methods are valid functions", () => {
```
Maybe we want to generate these tests based on the contents of `methodsWithCustomTypes` from `src/typed-method-types/mod.ts`? Rather than hard-coding the list in both files?

filmaj reviewed (commented):
Left one comment on the code, and not entirely sure how to test this, but here's what I did:

- Created a new reverse string app using the Slack CLI.
- Checked out this branch and ran `script/generate` from this repo. I just followed the instructions from the maintainers guide. Hope those steps are correct as I'm not familiar with this area much!
- Updated my app's import_map to point the deno-slack-api dependency to my local deno-slack-api repo with this branch checked out.
- Added a call to `client.workflows.triggers.create` in my fresh app's reverse string function. Saw an error for the argument into it - if I provide no argument, deno complains that I must provide one, and when I do provide one, it says the argument type is not assignable to `never`.
- The object that `triggers.create` returns is of type `BaseResponse`, with no trigger-specific methods available via autocompletion.

And just for the record, this is not isolated to my specific IDE, I also verified the same errors exist in VSCode 😉 

<img width="660" alt="Screen Shot 2022-07-14 at 9 29 52 AM" src="https://user-images.githubusercontent.com/52645/178994964-0a0f415d-2941-4376-85b5-04da1be93834.png">
<img width="380" alt="Screen Shot 2022-07-14 at 9 30 31 AM" src="https://user-images.githubusercontent.com/52645/178995073-aac38337-caf0-441e-9e3e-90fc44b112af.png">
<img width="333" alt="Screen Shot 2022-07-14 at 9 30 14 AM" src="https://user-images.githubusercontent.com/52645/178995155-9a41d1f8-9d14-45ab-a504-f10575295408.png">

shapirone commented:
Thanks @filmaj! Looks like a last minute rename caused the problem where the type was being registered as `never`. This has been fixed.

To add a new method with custom typing, you do not need to run the generate script. I added that loop to remove custom typed methods from the set in order to make sure we don't overload if we do have them in both, but I think leaving those methods in the array caused some of this confusion so I removed them. Since we don't need to run the generate script for this process unless you're replacing an existing method that was generated, I don't think there is a space to automate adding the tests.

Responses will come in a follow-up PR.
